metricflow/specs.py

```diff
@@ -346,10 +346,19 @@ class MetricInputMeasureSpec(SerializableDataclass):
     causes lookups connecting query -> data source to fail in strange ways. This spec, then, provides
     both the key (in the form of a MeasureSpec) along with whatever measure-specific attributes
     a user might specify in a metric definition or query accessing the metric itself.
+
+    Note - when specifying a metric comprised of two input instances of the same measure, at least one
+    must have a distinct alias, otherwise SQL exceptions may occur. This should be enforced via validation.
     """
 
     measure_spec: MeasureSpec
     constraint: Optional[SpecWhereClauseConstraint] = None
+    alias: Optional[str] = None
+
+    @property
+    def post_aggregation_spec(self) -> MeasureSpec:
+        """Return a MeasureSpec instance representing the post-aggregation spec state for the underlying measure"""
+        return MeasureSpec(element_name=self.alias) if self.alias else self.measure_spec
```

Comment:
what happens here for measures with non-additive-dimension? or is this what you mean by this approach being brittle and having to pass everything through the dataflow plan?

Reply:
Oh wow, I missed this. The new MeasureSpec needs to inherit the properties of the old one, just with a new name. I'll fix that here but I think we need to start thinking about factory methods for these "modified copy" cases so this logic isn't spread out all over the codebase.